src/main/java/org/zalando/nakadi/service/EventPublisher.java

```diff
@@ -114,14 +123,18 @@ private void partition(final List<BatchItem> batch, final EventType eventType) t
     }
 
     private void validate(final List<BatchItem> batch, final EventType eventType) throws EventValidationException,
-            InternalNakadiException {
+            EventSizeValidationException, InternalNakadiException {
         for (final BatchItem item : batch) {
             item.setStep(EventPublishingStep.VALIDATING);
             try {
                 validateSchema(item.getEvent(), eventType);
+                validateEventSize(item);
             } catch (final EventValidationException e) {
                 item.updateStatusAndDetail(EventPublishingStatus.FAILED, e.getMessage());
                 throw e;
+            } catch (final EventSizeValidationException e) {
+                item.updateStatusAndDetail(EventPublishingStatus.ABORTED, e.getMessage());
```

Comment:
https://github.com/zalando/nakadi/commit/957938b9bb4fa56ad95daeff1ea4d9c5d847e56f and https://github.com/zalando/nakadi/commit/49c184652e46d3748b1a8067d55297ec07c48fec